findfile.go

```diff
@@ -0,0 +1,158 @@
+	return &FindFile{fileName: fileName}
+}
+
+func (self *FindFile) Close() error {
+	if self.handle != 0 {
+		if err := wrappers.FindClose(self.handle); err != nil {
+			return NewWindowsError("FindClose", err)
+		}
+		self.handle = 0
+		wrappers.RtlZeroMemory((*byte)(unsafe.Pointer(&self.current)), unsafe.Sizeof(self.current))
+	}
+	return nil
+}
+
+func (self *FindFile) Current() FindFileItem {
+	return FindFileItem{
+		FileAttributes:    FileAttributes(self.current.FileAttributes),
+		FileSize:          (uint64(self.current.FileSizeHigh) << 32) | uint64(self.current.FileSizeLow),
+		FileName:          syscall.UTF16ToString(self.current.FileName[:]),
+		AlternateFileName: syscall.UTF16ToString(self.current.AlternateFileName[:]),
+	}
+}
+
+func (self *FindFile) Next() (bool, error) {
+	if self.handle == 0 {
+		handle, err := wrappers.FindFirstFile(syscall.StringToUTF16Ptr(self.fileName), &self.current)
+		if err == wrappers.ERROR_FILE_NOT_FOUND {
+			return false, nil
+		} else if err != nil {
+			return false, NewWindowsError("FindFirstFile", err)
+		}
+		self.handle = handle
+	} else {
+		if err := wrappers.FindNextFile(self.handle, &self.current); err == wrappers.ERROR_NO_MORE_FILES {
+			return false, nil
+		} else if err != nil {
+			return false, NewWindowsError("FindNextFile", err)
+		}
+	}
+	return true, nil
+}
+
+func GetDirectorySize(dirName string) (uint64, error) {
+	wildcard := filepath.Join(dirName, "*.*")
+	ff := OpenFindFile(wildcard)
+	defer ff.Close()
+	var totalSize uint64
+	for {
+		if more, err := ff.Next(); err != nil {
+			return 0, err
+		} else if !more {
+			break
+		}
+		info := ff.Current()
+		if info.FileName == "." || info.FileName == ".." {
+			continue
+		} else if (info.FileAttributes & FileAttributeDirectory) != 0 {
+			subdir := filepath.Join(dirName, info.FileName)
+			subdirSize, err := GetDirectorySize(subdir)
+			if err != nil {
+				return 0, err
+			}
+			totalSize += subdirSize
+		} else {
+			totalSize += info.FileSize
+		}
+	}
+	return totalSize, nil
+}
+
+func getDirectorySizeOnDisk(dirName string, clusterSize uint64) (uint64, error) {
+	wildcard := filepath.Join(dirName, "*.*")
+	ff := OpenFindFile(wildcard)
+	defer ff.Close()
+	var totalSize uint64
+	for {
+		if more, err := ff.Next(); err != nil {
+			return 0, err
+		} else if !more {
+			break
+		}
+		info := ff.Current()
+		if info.FileName == "." || info.FileName == ".." {
+			continue
+		} else if (info.FileAttributes & FileAttributeDirectory) != 0 {
+			subdir := filepath.Join(dirName, info.FileName)
+			subdirSize, err := getDirectorySizeOnDisk(subdir, clusterSize)
+			if err != nil {
+				return 0, err
+			}
+			totalSize += subdirSize
+		} else if (info.FileAttributes & FileAttributeCompressed) != 0 {
+			fileName := filepath.Join(dirName, info.FileName)
+			compressedSize, err := GetCompressedSize(fileName)
+			if err != nil {
+				return 0, err
+			}
+			totalSize += compressedSize
+		} else if info.FileSize%clusterSize != 0 {
```

Comment:
You are right. Put
info.FileSize%clusterSize
in parenthesis, and gofmt will parse it better.